---
title: "HMMs"
type: "manual"
menu:
  manual:
    parent: "Reference"
    weight: 80
---

# Introduction {#introduction}

Profile hidden Markov models (HMM) are used in Virtool to discover potential novel viruses in sequencing data. The models are used to identify known viral motifs in translated open reading frames derived from sequencing data. The method depends on our [NuVs](https://github.com/virtool/nuvs) bioinformatic pipeline.

In Virtool, profile HMMs are stored in the file `<data_path>/hmm/profiles.hmm`. This file is generated by [HMMER](http://hmmer.org), a piece of software used for generating and searching with profile HMMs. The `profiles.hmm` file contains a number of models generated from clustered amino acid sequences sourced from GenBank.

To make Virtool as easy to use as possible, we distribute an official set of models derived from the [vFam](http://derisilab.ucsf.edu/software/vFam/) project. Since these models do not carry biological annotations, we also provide a comprehensive set of annotations for these models calculated from metadata associated with the amino acid sequences used to build the models.

Offical HMM file releases can be manually downloaded from [this repository](https://github.com/virtool/virtool-hmm).

# Using Official Models {#official_models}

We highly recommend using the official models and annotations. It is very easy to obtain and use this data without leaving Virtool.

1. Go to _HMMs_ in the main navigation bar.

2. If there are no annotations and `profiles.hmm` does not exist, you will see the following:
   !["No HMM Data Found](no_data.png)

3. Click the <i class="vtfont i-download"></i> **Download** button to begin downloading the data. This could take some time.
   !["Installing HMMs](installing.png)

4. When the download is complete, the data will be imported into Virtool and you can browse the annotations and use the NuVs analysis algorithm.
   !["Browse HMMs"](top.png)

# Browsing Annotations {#browsing_annotations}

You can browse the annotations used in NuVs results.

1. Go to _HMMs_ in the main navigation bar.

2. Use the search bar to find HMMs by annotative name (_eg_. RNA replicase).

3. Click on an HMM item to view its details.

4. You will see the following view:
